### PR TITLE
Support for year-round heating/cooling unavailability

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

In the BuildResidentialHPXML measure, replaces:
* `schedules_vacancy_periods`
* `schedules_power_outage_periods`
* `schedules_power_outage_periods_window_natvent_availability`

With:
* `schedules_unavailable_period_types`
* `schedules_unavailable_period_dates`
* `schedules_unavailable_period_window_natvent_availabilities`

Also updates `unavailable_periods.csv` with:
* new column names `No Space Heating` and `No Space Cooling`
* replace row `hvac` with rows `space_heating` and `space_cooling`

Space heating and cooling can now be controlled independently.

## Checklist

Not all may apply:

- [ ] ~Schematron validator (`EPvalidator.xml`) has been updated~
- [x] Sample files have been added/updated (`openstudio tasks.rb update_hpxmls`)
- [ ] ~Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests/test*.rb` and/or `workflow/tests/test*.rb`)~
- [x] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -72,19 +72,19 @@ def arguments(model) # rubocop:disable Lint/UnusedMethodArgument
     arg.setDescription('Absolute/relative paths of csv files containing user-specified detailed schedules. If multiple files, use a comma-separated list.')
     args << arg
 
-    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_vacancy_periods', false)
-    arg.setDisplayName('Schedules: Vacancy Periods')
-    arg.setDescription('Specifies the vacancy periods. Enter a date like "Dec 15 - Jan 15". Optionally, can enter hour of the day like "Dec 15 2 - Jan 15 20" (start hour can be 0 through 23 and end hour can be 1 through 24). If multiple periods, use a comma-separated list.')
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_unavailable_period_types', false)
+    arg.setDisplayName('Schedules: Unavailable Period Types')
+    arg.setDescription('Specifies the unavailable period types. Possible types are column names defined in unavailable_periods.csv. If multiple periods, use a comma-separated list.')
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:77]: Can you parse the file and provide the user with the possible choices?

File: BuildResidentialHPXML/measure.rb [hunk 2]
```diff
@@ -72,19 +72,19 @@ def arguments(model) # rubocop:disable Lint/UnusedMethodArgument
...
     args << arg
 
-    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_power_outage_periods', false)
-    arg.setDisplayName('Schedules: Power Outage Periods')
-    arg.setDescription('Specifies the power outage periods. Enter a date like "Dec 15 - Jan 15". Optionally, can enter hour of the day like "Dec 15 2 - Jan 15 20" (start hour can be 0 through 23 and end hour can be 1 through 24). If multiple periods, use a comma-separated list.')
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_unavailable_period_dates', false)
+    arg.setDisplayName('Schedules: Unavailable Period Dates')
+    arg.setDescription('Specifies the unavailable period date ranges. Enter a date range like "Dec 15 - Jan 15". If multiple periods, use a comma-separated list. Optionally, can enter hour of the day like "Dec 15 2 - Jan 15 20" (start hour can be 0 through 23 and end hour can be 1 through 24). If multiple periods, use a comma-separated list.')
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:82]: "If multiple periods, use a comma-separated list." is mentioned twice, remove the first instance.

File: BuildResidentialHPXML/tests/test_build_residential_hpxml.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@
 class BuildResidentialHPXMLTest < Minitest::Test
   def setup
     @output_path = File.join(File.dirname(__FILE__), 'extra_files')
-    @model_save = false # true helpful for debugging, i.e., can render osm in 3D
+    @model_save = true # true helpful for debugging, i.e., can render osm in 3D
```
Comment (shorowit) [inline BuildResidentialHPXML/tests/test_build_residential_hpxml.rb:13]: Was this a temporary change?

File: HPXMLtoOpenStudio/measure.rb [hunk 4]
```diff
@@ -525,7 +525,9 @@ def create_unit_model(hpxml, hpxml_bldg, runner, model, epw_path, weather, debug
     add_num_occupants(model, runner, spaces)
 
     # HVAC
-    @hvac_unavailable_periods = Schedule.get_unavailable_periods(runner, SchedulesFile::Columns[:HVAC].name, @hpxml_header.unavailable_periods)
+    @hvac_unavailable_periods = []
+    @hvac_unavailable_periods << Schedule.get_unavailable_periods(runner, SchedulesFile::Columns[:SpaceHeating].name, @hpxml_header.unavailable_periods)
+    @hvac_unavailable_periods << Schedule.get_unavailable_periods(runner, SchedulesFile::Columns[:SpaceCooling].name, @hpxml_header.unavailable_periods)
```
Comment (shorowit) [inline HPXMLtoOpenStudio/measure.rb:530]: Why not just create `heating_unavailable_periods` and `cooling_unavailable_periods`? An array feels overkill. And then you don't have to `flatten` below.

File: HPXMLtoOpenStudio/measure.rb [hunk 5]
```diff
@@ -2283,8 +2285,9 @@ def add_airflow(runner, model, weather, spaces, airloop_map)
 
     # Create HVAC availability sensor
     hvac_availability_sensor = nil
+    @hvac_unavailable_periods = @hvac_unavailable_periods.flatten
     if not @hvac_unavailable_periods.empty?
-      avail_sch = ScheduleConstant.new(model, SchedulesFile::Columns[:HVAC].name, 1.0, EPlus::ScheduleTypeLimitsFraction, unavailable_periods: @hvac_unavailable_periods)
+      avail_sch = ScheduleConstant.new(model, 'hvac', 1.0, EPlus::ScheduleTypeLimitsFraction, unavailable_periods: @hvac_unavailable_periods)
```
Comment (shorowit) [inline HPXMLtoOpenStudio/measure.rb:2290]: What is this about?

File: workflow/hpxml_inputs.json [hunk 6]
```diff
@@ -1003,7 +1003,9 @@
   },
   "sample_files/base-battery-scheduled-power-outage.xml": {
     "parent_hpxml": "sample_files/base-battery-scheduled.xml",
-    "schedules_power_outage_periods": "Jul 1 5 - Jul 31 14"
+    "schedules_unavailable_period_types": "Power Outage",
+    "schedules_unavailable_period_dates": "Jul 1 5 - Jul 31 14",
+    "schedules_unavailable_period_window_natvent_availabilities": "regular schedule"
```
Comment (shorowit) [inline workflow/hpxml_inputs.json:1008]: Why do you have to provide the `window_natvent_availabilities` argument now? What was it doing before?

File: workflow/sample_files/base-battery-scheduled-power-outage.xml [hunk 7]
```diff
@@ -25,6 +25,7 @@
           <EndMonth>7</EndMonth>
           <EndDayOfMonth>31</EndDayOfMonth>
           <EndHourOfDay>14</EndHourOfDay>
+          <NaturalVentilation>regular schedule</NaturalVentilation>
```
Comment (shorowit) [inline workflow/sample_files/base-battery-scheduled-power-outage.xml:28]: Ideally these files wouldn't change.

**Review by shorowit — COMMENTED** [review]
The overall approach looks reasonable, but I did have a few questions/suggestions.

File: HPXMLtoOpenStudio/measure.rb (hunk 5 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/measure.rb:2290]: The `HVAC` key was replaced by `SpaceHeating` and `SpaceCooling`. But we still want to create a constant schedule (1s) with unavailable period rules (0s) that we can put an EMS sensor on.

File: workflow/hpxml_inputs.json (hunk 6 above)
Comment (joseph-robertson) [reply workflow/hpxml_inputs.json:1008]: I'm now requiring that the 3 new arguments are either (1) all not initialized or (2) all initialized. And, if `schedules_unavailable_period_window_natvent_availabilities` is initialized, it must contain only valid ventilation types.

File: workflow/sample_files/base-battery-scheduled-power-outage.xml (hunk 7 above)
Comment (joseph-robertson) [reply workflow/sample_files/base-battery-scheduled-power-outage.xml:28]: See response above.

(I suppose we could relax `schedules_unavailable_period_window_natvent_availabilities ` being required if the other 2 arguments are initialized. We could also relax that its elements are all valid ventilation types, e.g., allow `,,regular schedule,,,always unavailable,,,,`.)

File: HPXMLtoOpenStudio/measure.rb (hunk 5 above)
Comment (shorowit) [reply HPXMLtoOpenStudio/measure.rb]: I see. I guess it's fine. I would have thought the name should be more descriptive, but I guess it was simply "hvac" before?

File: workflow/sample_files/base-battery-scheduled-power-outage.xml (hunk 7 above)
Comment (shorowit) [reply workflow/sample_files/base-battery-scheduled-power-outage.xml:28]: Yes, I'm in favor of allowing this argument to be optional. If there are, say, 3 comma-separated values provided to the other two arguments, It'd be nice if this argument could either be `"foo,foo,foo"` or `",,foo"` or even just empty `""`.

File: HPXMLtoOpenStudio/measure.rb (hunk 5 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/measure.rb]: Yes, it was "hvac" before. I can update it to be a bit more descriptive.

**Review by shorowit — APPROVED** [review]
LGTM. Assuming I didn't break anything in my last commit, this can go in. Thanks @joseph-robertson.

**Pull request merged**